src/Sulu/Bundle/SecurityBundle/Resources/public/js/services/role-manager.js

```diff
@@ -0,0 +1,186 @@
+        };
+
+    /** @constructor **/
+    function RoleManager() {
+    }
+
+    RoleManager.prototype = {
+
+        /**
+         * Load or create a role by given id
+         * @param {undefined|Number} roleId The role id to load the role for
+         * @returns {Object} a promise
+         */
+        loadOrNew: function(roleId) {
+            var promise = $.Deferred(),
+                role;
+
+            if (!roleId) {
+                role = new Role();
+                promise.resolve(role.toJSON());
+            } else {
+                role = Role.findOrCreate({id: roleId});
+                role.fetch({
+                    success: function() {
+                        Mediator.emit('sulu.security.role.loaded', roleId);
+                        promise.resolve(role.toJSON());
+                    }.bind(this),
+                    error: function() {
+                        promise.reject();
+                    }.bind(this)
+                });
+            }
+
+            return promise;
+        },
+
+        /**
+         * Saves a role by given data. Iff no id is set,
+         * a new role gets created.
+         *
+         * @param {Object} data The data to save for the role.
+         * @returns {Object} a promise
+         */
+        save: function(data) {
+            var promise = $.Deferred();
+
+            saveRole(data).done(function(role) {
+                Mediator.emit('sulu.security.role.saved', role.id);
+                Mediator.emit('sulu.labels.success.show', 'sulu-security.role.saved');
+                Mediator.emit('sulu.labels.warning.show', 'security.warning');
+                promise.resolve(role);
+            }.bind(this)).fail(function(response) {
+                Mediator.emit('sulu.labels.error.show', getErrorStringForCode(response.responseJSON.code));
+                promise.reject();
+            }.bind(this));
+
+            return promise;
+        },
+
+        /**
+         * Deletes one or more roles
+         *
+         * @param {Array|Number} roleIds The id or an array of ids to delete
+         * @returns {Object} a promise which gets resolved when all roles got deleted
+         */
+        delete: function(roleIds) {
+            if (!$.isArray(roleIds)) {
+                roleIds = [roleIds];
+            }
+
+            var requests = [],
+                promise = $.Deferred();
+
+            Util.each(roleIds, function(index, id) {
+                requests.push(deleteRole(id));
+            }.bind(this));
+
+            $.when.apply(null, requests).done(function() {
+                promise.resolve();
+            }.bind(this)).fail(function() {
+                promise.reject();
+            }.bind(this));
+
+            return promise;
+        }
+    };
+
+    RoleManager.getInstance = function() {
+        if (instance === null) {
+            instance = new RoleManager();
+        }
+        return instance;
+    };
+
+    return RoleManager.getInstance();
```

Comment:
A component already is a singleton, so there is actually no need to implement it again (no `instance` variable, no `getInstance` method, no prototypes, simply return an object here.